proto/redpanda/runtime/v1alpha1/input.proto

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package redpanda.runtime.v1alpha1;
+
+option go_package = "github.com/redpanda-data/connect/v4/internal/dynamic/plugin/runtimepb";
+
+import "redpanda/runtime/v1alpha1/message.proto";
+
+
+// BatchInput is an interface implemented by Benthos inputs that produce messages
+// in batches, where there is a desire to process and send the batch as a logical
+// group rather than as individual messages.
+// 
+// Calls to ReadBatch should block until either a message batch is ready to process,
+// the connection is lost, or the provided context is cancelled.
+service BatchInputService {
+  // Init is the first method called for a batch input and it passes the user's
+  // configuration to the input.
+  //
+  // The schema for the input configuration is specified in the `plugin.yaml` file
+  // provided to Redpanda Connect.
+  rpc Init(InputInitRequest) returns (InputInitResponse);
+  // Establish a connection to the upstream service. Connect will always be
+  // called first when a reader is instantiated, and will be continuously
+  // called with back off until a nil error is returned.
+  //
+  // The provided context remains open only for the duration of the connecting
+  // phase, and should not be used to establish the lifetime of the connection
+  // itself.
+  //
+  // Once Connect returns a nil error the Read method will be called until
+  // either ErrNotConnected is returned, or the reader is closed.
+  rpc Connect(InputConnectRequest) returns (InputConnectResponse);
+  // Read a message batch from a source, along with a function to be called
+  // once the entire batch can be either acked (successfully sent or
+  // intentionally filtered) or nacked (failed to be processed or dispatched
+  // to the output).
+  //
+  // The Ack will be called for every message batch at least once, but
+  // there are no guarantees as to when this will occur. If your input
+  // implementation doesn't have a specific mechanism for dealing with a nack
+  // then you can instruct the Connect framework to auto_replay_nacks in the 
+  // InitResponse to get automatic retries.
+  //
+  // If this method returns Error.NotConnected then ReadBatch will not be called
+  // again until Connect has returned a nil error. If Error.EndOfInput is
+  // returned then Read will no longer be called and the pipeline will
+  // gracefully terminate.
+  rpc ReadBatch(InputReadRequest) returns (InputReadResponse);
```

Comment:
Consider renaming the types used with BatchInputService to use BatchInput prefix ex. BatchInputReadRequest vs. InputReadRequest.

Reply:
Done